doc/lsst.pipe.tasks/deblending-flags-overview.rst

```diff
@@ -0,0 +1,153 @@
+
+Multi-band deblender (`scarlet`)
+================================
+
+:py:class:`lsst.meas.extensions.scarlet.ScarletDeblendTask` is different, as it uses `scarlet` to create a *model* for each source in a blend. This is a philosophically different object, as there is no longer an assumption that all of the flux in the input image will be modeled by one of the children in the blend (this may change in the near future, but this is the current implementation). The results of the scarlet deblender will be biased by the assumptions that went into making the models, so it was decided that it would be a good idea to (by default) also model all of the isolated sources. This will allow comparisons of scarlet models of isolated sources to the un-modeled isolated source measurements to investigate the biases that scarlet is introducing and also gives users the option to choose between the un-modeled (parent) isolated source records and the scarlet model version of each isolated source. However this flexibility forced a change in the way that we select unique objects in a source catalog.
+
+Flags set by the deblender
+==========================
+
+Before we get into the flags set by pipe_tasks it is useful to understand the flags that are set in `SourceDeblendTask` and `ScarletDeblendTask` that relate to source selection.
+
+- `parent`: the id in the catalog for the parent of this source record. This is actually set pre-deblender, where all top level records have `parent=0`.
+- `deblend_nPeaks`: the number of peaks contained in the sources footprint.
+- `deblend_nChild`: the number of peaks deblended by the deblender from this source and created as new source records in the catalog. This is different from `deblend_nPeaks` in that isolated sources that are not deblended by `SourceDeblendTask` and child peaks that were culled during deblending are not included in this count.
+- `deblend_parentNPeaks`: The number of peaks contained in the parent of this source record.
+- `deblend_parentNChild`: the number of children deblended from the parent of this source record.
+
+`isPrimary` and other flags added in `pipe_tasks`
+=================================================
+
+In addition to source records for deblended parents and multiple entries for isolated sources, output catalogs are also not unique because they may contain "pseduo" sources (eg. sky objects that have been added to assist with calibration but are not output sources) and, if the analysis is done over multiple patches and/or tracts, sources in the overlap region can exist in multiple overlapping patches (but always on the interior of only one). For this reason the :py:class:`lsst.pipe.tasks.SetPrimaryFlagsTask` sets a number of useful flags to assist users in determining a unique output catalog for their analysis.
```

Comment:
```suggestion
In addition to source records for deblended parents and multiple entries for isolated sources, output catalogs are also not unique because they may contain "pseduo" sources (eg. sky objects that have been added to assist with calibration but are not output sources) and, if the analysis is done over multiple patches and/or tracts, sources in the overlap region can exist in multiple overlapping patches (but always on the interior of only one). For this reason the :lsst-task:`~lsst.pipe.tasks.SetPrimaryFlagsTask` task sets a number of useful flags to help you determine a unique output catalog for your analysis.
```

- Using the `lsst-task` role to link to the task reference page
- Fixing the third-person tense.